mypy/fastparse.py

```diff
@@ -769,139 +781,140 @@ def visit_UnaryOp(self, n: ast3.UnaryOp) -> UnaryExpr:
-                             targets,
-                             iters,
-                             ifs_list,
-                             is_async)
+        e = GeneratorExpr(self.visit(n.elt),
+                          targets,
+                          iters,
+                          ifs_list,
+                          is_async)
+        return self.set_line(e, n)
 
     # Await(expr value)
-    @with_line
     def visit_Await(self, n: ast3.Await) -> AwaitExpr:
         v = self.visit(n.value)
-        return AwaitExpr(v)
+        e = AwaitExpr(v)
+        return self.set_line(e, n)
 
     # Yield(expr? value)
-    @with_line
     def visit_Yield(self, n: ast3.Yield) -> YieldExpr:
-        return YieldExpr(self.visit(n.value))
+        e = YieldExpr(self.visit(n.value))
+        return self.set_line(e, n)
 
     # YieldFrom(expr value)
-    @with_line
     def visit_YieldFrom(self, n: ast3.YieldFrom) -> YieldFromExpr:
-        return YieldFromExpr(self.visit(n.value))
+        e = YieldFromExpr(self.visit(n.value))
+        return self.set_line(e, n)
 
     # Compare(expr left, cmpop* ops, expr* comparators)
-    @with_line
     def visit_Compare(self, n: ast3.Compare) -> ComparisonExpr:
         operators = [self.from_comp_operator(o) for o in n.ops]
         operands = self.translate_expr_list([n.left] + n.comparators)
-        return ComparisonExpr(operators, operands)
+        e = ComparisonExpr(operators, operands)
+        return self.set_line(e, n)
 
     # Call(expr func, expr* args, keyword* keywords)
     # keyword = (identifier? arg, expr value)
-    @with_line
-    def visit_Call(self, n: ast3.Call) -> CallExpr:
-        def is_star2arg(k: ast3.keyword) -> bool:
-            return k.arg is None
-
+    def visit_Call(self, n: Call) -> CallExpr:
+        args = n.args
+        keywords = n.keywords
+        keyword_names = [k.arg for k in keywords]
         arg_types = self.translate_expr_list(
-            [a.value if isinstance(a, ast3.Starred) else a for a in n.args] +
-            [k.value for k in n.keywords])
-        arg_kinds = ([ARG_STAR if isinstance(a, ast3.Starred) else ARG_POS for a in n.args] +
-                     [ARG_STAR2 if is_star2arg(k) else ARG_NAMED for k in n.keywords])
-        return CallExpr(self.visit(n.func),
-                        arg_types,
-                        arg_kinds,
-                        cast(List[Optional[str]], [None] * len(n.args)) +
-                        [k.arg for k in n.keywords])
+            [a.value if isinstance(a, Starred) else a for a in args] +
+            [k.value for k in keywords])
+        arg_kinds = ([ARG_STAR if type(a) is Starred else ARG_POS for a in args] +
+                     [ARG_STAR2 if arg is None else ARG_NAMED for arg in keyword_names])
+        e = CallExpr(self.visit(n.func),
+                     arg_types,
+                     arg_kinds,
+                     cast('List[Optional[str]]', [None] * len(args)) + keyword_names)
```

Comment:
Why the quotes are needed around `List[Optional[str]]` now?

Reply:
They are not needed but they speed things up on older Python versions (when not compiled).